### DISCUSSION TITLE
Is it safe from log4j zero day exploit?

### DISCUSSION THREAD

**Discussion by m0rph2us** [keycloak/keycloak#9065] [Q&A] [answered]
Hi guys

Is it safe from log4j zero day exploit(https://www.lunasec.io/docs/blog/log4j-zero-day/)?

**Comment by piec** [comment]
Hi,

**Edit: please see the answers from the keycloak team https://github.com/keycloak/keycloak/discussions/9065#discussioncomment-1786130**

I'm wondering the same.
~~I would say that keycloak is not vulnerable because it seems to use [a 1.X version of log4j](https://github.com/keycloak/keycloak/blob/7aaa33739bc1543c4ef61ebbc993b24c7f0b1416/pom.xml#L94).~~
~~Affected version are [`2.0 <= Apache log4j <= 2.14.1`](https://www.lunasec.io/docs/blog/log4j-zero-day/#affected-apache-log4j-versions).~~

Still it has to be confirmed by the keycloak team.

**Comment by stianst** [comment]
The initial assessment here is that Keycloak is not affected by this issue, but we are still looking into it in more detail.

Keycloak does not use log4j-core, but rather just has a Log4j facade onto JBoss Logging manager.

**Reply by Joni-Aaltonen** [reply]
Thanks for the reply. JBoss Logging [included in keycloak](https://github.com/keycloak/keycloak/blob/7aaa33739bc1543c4ef61ebbc993b24c7f0b1416/pom.xml#L88) does however include an affected version of log4j (https://github.com/jboss-logging/jboss-logging/blob/master/pom.xml#L36).

Wether that used or not is a different case, [by default it seems to be using JBoss LogManager](https://docs.jboss.org/hibernate/orm/4.3/topical/html/logging/Logging.html).

**Reply by stianst** [reply]
Update on that one; and JBoss LogManager does include potentially affected classes from Log4j within its jar. We are still investigating and will provide more details as soon as we can.

**Comment by ibauersachs** [comment]
According to https://github.com/apache/logging-log4j2/pull/608#issuecomment-990494126 the old log4j (v1) is also affected. I'm not sure if or how that affects Keycloak.

**Reply by Joni-Aaltonen** [reply]
[Different vulnerability](https://www.cvedetails.com/cve/CVE-2019-17571/), but your point is still very valid as 1.x is EOL.

**Comment by stianst** [comment, marked as answer]
We've reached a conclusion and Keycloak is NOT affected by this vulnerability. For more details see https://github.com/keycloak/keycloak/discussions/9078
